qutip/cy/dicke.pyx

```diff
@@ -0,0 +1,645 @@
+        """
+        Calculate gamma3 for given j, m, m'
+        """
+        cdef float j, m, m1
+        j, m, m1 = jmm1
+
+        cdef float yE
+
+        cdef float N
+        N = float(self.N)
+
+        cdef float spontaneous, losses, pump, collective_pump
+        cdef float dephase, collective_dephase
+
+        cdef complex g3
+
+        yE = self.emission
+
+        if (yE == 0) or (j <= 0):
+            g3 = 0.0
+        else:
+            g3 = yE / 2 * np.sqrt((j + m) * (j + m - 1) * (j + m1) * (j + m1 - 1)) * (N / 2 + j + 1) / (j * (2 * j + 1))
+
+        return (g3)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cpdef complex gamma4(self, tuple jmm1):
+        """
+        Calculate gamma4 for given j, m, m'
+        """
+        cdef float j, m, m1
+        j, m, m1 = jmm1
+
+        cdef float yE
+        cdef float N
+        N = float(self.N)
+
+        cdef complex g4
+
+        yE = self.emission
+
+        if (yE == 0) or ((j + 1) <= 0):
+            g4 = 0.0
+        else:
+            g4 = yE / 2 * np.sqrt((j - m + 1) * (j - m + 2) * (j - m1 + 1) * (j - m1 + 2)) * (N / 2 - j) / ((j + 1) * (2 * j + 1))
+
+        return (g4)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cpdef complex gamma5(self, tuple jmm1):
+        """
+        Calculate gamma5 for given j, m, m'
+        """
+        cdef float j, m, m1
+        j, m, m1 = jmm1
+
+        cdef float yD
+        cdef float N
+        N = float(self.N)
+
+        cdef complex g5
+
+        yD = self.dephasing
+
+        if (yD == 0) or (j <= 0):
+            g5 = 0.0
+        else:
+            g5 = yD / 2 * np.sqrt((j**2 - m**2) * (j**2 - m1**2)) * \
+                (N / 2 + j + 1) / (j * (2 * j + 1))
+
+        return (g5)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cpdef complex gamma6(self, tuple jmm1):
+        """
+        Calculate gamma6 for given j, m, m'
+        """
+        cdef float j, m, m1
+        j, m, m1 = jmm1
+
+        cdef float yD
+        cdef float N
+        N = float(self.N)
+
+        cdef complex g6
+
+        yD = self.dephasing
+
+        if yD == 0:
+            g6 = 0.0
+        else:
+            g6 = yD / 2 * np.sqrt(((j + 1)**2 - m**2) * ((j + 1) **
+                                                         2 - m1**2)) * (N / 2 - j) / ((j + 1) * (2 * j + 1))
+
+        return (g6)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cpdef complex gamma7(self, tuple jmm1):
+        """
+        Calculate gamma7 for given j, m, m'
+        """
+        cdef float j, m, m1
+        j, m, m1 = jmm1
+
+        cdef float yP
+        cdef float N
+        N = float(self.N)
+
+        cdef complex g7
+
+        yP = self.pumping
+
+        if (yP == 0) or (j <= 0):
+            g7 = 0.0
+        else:
+            g7 = yP / 2 * np.sqrt((j - m - 1) * (j - m) * (j - m1 - 1) * (j - m1)) * (N / 2 + j + 1) / (j * (2 * j + 1))
```

Comment:
You could probably shorten this line by removing what I would consider some unnecessary spaces, e.g.
```
(N/2 + j + 1) / (j*(2*j + 1))
```
Not everyone agrees on this kind of thing though. I suggest whatever you think looks most clear in terms of precedence